includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -1055,6 +1055,67 @@ public function generate_create_intent_request( $order, $prepared_source ) {
 		return apply_filters( 'wc_stripe_generate_create_intent_request', $request, $order, $prepared_source );
 	}
 
+	/**
+	 * Create the level 3 data array to send to Stripe when making a purchase.
+	 *
+	 * @param WC_Order $order The order that is being paid for.
+	 * @return array          The level 3 data to send to Stripe.
+	 */
+	public function get_level3_data_from_order( $order ) {
+		// Return an empty array if there's no store postcode (since it's
+		// required for level3 data), or if the version is less than 3.0
+		// (because it's not possible to get a store postcode before WC 3.0).
+		if ( WC_Stripe_Helper::is_wc_lt( '3.0' ) ) {
+			return array();
+		}
+
+		// Get the order items. Don't need their keys, only their values.
+		// Order item IDs are used as keys in the original order items array.
+		$order_items = array_values( $order->get_items() );
+		$currency    = $order->get_currency();
+
+		$stripe_line_items = array_map(function( $item ) use ( $currency ) {
+			$product_id          = $item->get_variation_id()
+				? $item->get_variation_id()
+				: $item->get_product_id();
```

Comment:
Do we need to limit the `ID` here to 12 characters similar to what we do for `$product_description`, or are we certain the `ID` will always be 12 characters or less?

Reply:
The ID in this case is an integer, coming from the product/post ID. The DB column is a BIGINT(20), however I don't expect any issues with this. If someone's store has 999999999999+ posts, I think they have bigger problems to worry about.

Reply:
hahaha sounds about right 😛